src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -508,19 +510,48 @@ private void processConverterStation(LfNetwork lfNetwork, HvdcConverterStation<?
         }
     }
 
-    private Map<LfBus, BusState> applyHvdcLineContingency(Network network, LfNetwork lfNetwork, PropagatedContingency contingency) {
-        // it applies on the network the loss of the HVDC lines contained in the contingency.
-        // Buses state are stored too.
+    private void processGenerator(LfNetwork lfNetwork, Generator generator,
+                                         Collection<Pair<LfBus, LfGenerator>> generators) {
+        AbstractLfBus bus = (AbstractLfBus) lfNetwork.getBusById(generator.getTerminal().getBusView().getBus().getId());
+        if (bus == null) {
+            return;
+        } else {
+            generators.add(Pair.of(bus, bus.getGenerators().stream().filter(LfGenerator.class::isInstance)
+                    .map(LfGenerator.class::cast)
+                    .filter(lfGenerator -> lfGenerator.getId().equals(generator.getId()))
+                    .findFirst().orElseThrow()));
+        }
+    }
+
+    private Map<LfBus, BusState> applyContingencyOnNetwork(Network network, LfNetwork lfNetwork, PropagatedContingency contingency,
```

Comment:
The name is confusing as we only apply injection part of the contingency

Reply:
I have changed in `applyInjectionContingencies`, what do you think?